File: 1-1.rb
#!/usr/bin/env ruby

values = File.read('1.input').lines.map(&:to_i)

elf_calories = Hash.new 0

elf = 0
values.each do |v|
  if v.zero?
    elf += 1
  else
    elf_calories[elf] += v
  end
end

_, max = elf_calories.max_by { |_, c| c }

print max, "\n"

File: 1-1.py
#!/usr/bin/env python

values = [row.strip() for row in open('1.input').readlines()]

elf_calories = {}

elf = 0
for v in values:
    if v == '':
        elf += 1
    else:
        elf_calories[elf] = elf_calories.get(elf, 0) + int(v)

print(max(elf_calories.values()))
